corehq/apps/app_manager/tests/test_apps.py

```diff
@@ -136,10 +136,26 @@ def testDeleteForm(self):
         for module, i in zip(self.app.get_modules(), [2, 3, 3]):
             self.assertEqual(len(module.forms), i)
 
+    def test_undo_delete_form_removes_deleted_couch_doc_record(self):
+        form = self.app.modules[0].forms[0]
+        rec = self.app.delete_form(self.app.modules[0].unique_id, form.unique_id)
+        assert DeletedCouchDoc.objects.get(doc_id=rec._id, doc_type=rec.doc_type)
+        rec.undo()
+        with self.assertRaises(DeletedCouchDoc.DoesNotExist):
+            DeletedCouchDoc.objects.get(doc_id=rec._id, doc_type=rec.doc_type)
```

Comment:
Here is a safer pattern to avoid a hard-coded doc type literal, ensuring that the specific `DeletedCouchDoc` in question has been deleted regardless of what changes are made by `rec.undo()`.
```suggestion
        params = {"doc_id": rec._id, "doc_type": rec.doc_type}
        assert DeletedCouchDoc.objects.get(**params)
        rec.undo()
        with self.assertRaises(DeletedCouchDoc.DoesNotExist):
            DeletedCouchDoc.objects.get(**params)
```

Reply:
Made these changes to all the applicable tests!